server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -724,7 +724,8 @@ public QueryRewriteContext newQueryRewriteContext(
             namedWriteableRegistry,
             valuesSourceRegistry,
             allowExpensiveQueries,
-            scriptService
+            scriptService,
+            null
```

Comment:
`indexMetadataMapSupplier` is set to `null` here because a `QueryRewriteContext` will never be generated for the semantic query through this code path. 

However, I could set `indexMetadataMapSupplier` to a supplier that returns a single-entry map with the index metadata for the index handled by the `IndexService` instance, like so:

```
() -> Map.of(index().getName(), getIndexSettings().getIndexMetadata())
```

Reply:
I think it's a good distinction for now since this information is to be used on the coordinating node. We can cleanup when/if we introduce a real distinction between shard and coordinator rewrite context.